### PR TITLE
Fixes for NHL

### PR DESCRIPTION
**Pull request by cameronkeif**
Fixes a few bugs I noticed with the new NHL functionality from dcd323e0a85dc9d942dcb4374fc0cfe104660e38:

* Goalies were not being updated, because they are in a separate `tbody.Table__TBODY` element. This is also true for baseball pitchers if that's ever implemented.
* The extension was hardcoded against `F` slots, but did not work for separate forward positions (`C`, `LW`, `RW`).
* Unknown slot errors were caused by what looks some old code that was moved to the NBA config, but wasn't removed from the root `autostart.js` (the `parseSlotType(...)` function)

---
**Demo**

Single Day
hockey-fix

Full week
hockey-fix-week

### CODE REVIEW

**Comment by cameronkeif** [comment]
Sidenote: awesome extension :smile:

File: src/common/autosetup.js [hunk 1]
```diff
@@ -142,41 +142,26 @@ function goToNextDay(loadDelay, loadMaxAttempts, action) {
...
-        const slotType = parseSlotType(slotCell.textContent);
-        if (slotType === SLOT_TYPE_BENCH || slotType === SLOT_TYPE_IR) {
-            benchRows.push(row);
-        } else {
-            starterRows.push(row);
+    const playerTable1 = tableBody[0];
+    const playerTable2 = tableBody.length > 3 ? tableBody[3] : [];
+
+    [playerTable1, playerTable2].forEach((table) => {
+        for (const row of table.getElementsByTagName('tr')) {
```
Comment (magnusbakken) [inline src/common/autosetup.js]: This doesn't quite work for the NBA, where there's only one table. getElementsByTagName gives an error when the playerTable2 variable is an empty list.

This could be fixed in several ways. Probably the easiest way is to set playerTable2 to undefined/null instead of an empty list and test the table variable for that before iterating.

**Review by magnusbakken — CHANGES_REQUESTED** [review]
Seems to work fine apart from the empty table issue.

File: src/common/autosetup.js (hunk 1 above)
Comment (cameronkeif) [reply src/common/autosetup.js]: Ah whoops, I forgot that `tableBody`'s elements were `NodeList`s, not arrays! Gotta love cases where you can't access the page to test :laughing: .

I think I have fixed this in 8faf31a23ee09f215ed2d75805f67a064ae51b3f.

Comment (magnusbakken) [reply src/common/autosetup.js]: Yeah, it's an interesting way to write code. 😆

Testing this extension is a mess in general. Sometimes I have to simply wait for a certain combination of players and games to show up in order to test corner cases. Between seasons it usually breaks because the page layout changes.

It would probably be possible to set up some sort of test workflow with a mocked version of the HTML and server, but to be honest I don't want to put too much work into an extension that ESPN could so easily replace with a natively implemented version.

Anyway, the code seems to work just fine now, so I'll merge this!
